controller/api/destination/profile_translator.go

```diff
@@ -103,6 +106,22 @@ func (pt *profileTranslator) toServiceProfile(profile *sp.ServiceProfile) (*pb.D
 	if profile.Spec.OpaquePorts != nil {
 		_, opaqueProtocol = profile.Spec.OpaquePorts[pt.port]
 	}
+	// Only set the opaque transport if the translator has an endpoint with a
+	// protocol hint.
+	if pt.endpoint != nil && pt.endpoint.ProtocolHint != nil {
+		if !opaqueProtocol {
+			pt.endpoint.ProtocolHint.OpaqueTransport = nil
+		} else if pt.endpoint.ProtocolHint.OpaqueTransport == nil {
+			port, err := getInboundPort(&pt.pod.Spec)
```

Comment:
If `pt.endpoint != nil` and `pt.pod == nil`, this could hit a nil pointer deref.  Can we hit that case with a non-pod backed endpoint?